### PR TITLE
Internal: (infra) jQuery Migrate deprecation warnings [ED-13116]

### PR DESCRIPTION
**Pull request by mark-elementor**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

**Review by Omerisra6 — DISMISSED** [review]

**Comment by rami-elementor** [comment]
Search for `$.isFunction` in the entire project. You will find additional two more files that use this. May be we should fix them too?

**Review by rami-elementor — APPROVED** [review]

File: assets/lib/dialog/dialog.js [hunk 1]
```diff
@@ -880,7 +880,7 @@
 			}
 
 			if (this.focusedButton) {
-				this.focusedButton.focus();
+				this.focusedButton.trigger('focus');
```
Comment (StyleShit) [inline assets/lib/dialog/dialog.js:883]: shouldn't be fixed in https://github.com/kobizz/dialogs-manager?

Comment (rami-elementor) [reply assets/lib/dialog/dialog.js:883]: Correct. See: https://github.com/kobizz/dialogs-manager/pull/20

When the other library will be update I'll merge it to Elementor.

Meanwhile, @mark-elementor please remove the changes from the `dialog.js` file.

**Review by rami-elementor — CHANGES_REQUESTED** [review]

File: assets/lib/dialog/dialog.js (hunk 1 above)
Comment (rami-elementor) [reply assets/lib/dialog/dialog.js]: not relevant any more, the `dialog.js` file was already updated.

**Review by rami-elementor — DISMISSED** [review]

**Review by rami-elementor — APPROVED** [review]

**Review by Omerisra6 — DISMISSED** [review]

File: assets/lib/e-select2/js/e-select2.full.js [hunk 2]
```diff
@@ -3703,7 +3705,7 @@ S2.define('select2/data/tags',[
   };
 
   Tags.prototype.createTag = function (decorated, params) {
-    var term = $.trim(params.term);
+    var term = params.term.trim();
```
Comment (rami-elementor) [inline assets/lib/e-select2/js/e-select2.full.js:3708]:
```suggestion
    var term = params?.term?.trim();
```

**Review by rami-elementor — CHANGES_REQUESTED** [review]

File: assets/lib/e-select2/js/e-select2.full.js [hunk 3]
```diff
@@ -6098,7 +6100,7 @@ S2.define('select2/compat/matcher',[
     function wrappedMatcher (params, data) {
       var match = $.extend(true, {}, data);
 
-      if (params.term == null || $.trim(params.term) === '') {
+      if (params.term == null || params.term.trim() === '') {
```
Comment (Omerisra6) [inline assets/lib/e-select2/js/e-select2.full.js:6103]:
```suggestion
      if (params.term == null || params?.term?.trim() === '') {
```

File: assets/lib/e-select2/js/e-select2.full.js [hunk 4]
```diff
@@ -5780,7 +5782,7 @@ S2.define('select2/compat/utils',[
       });
     }
 
-    classes = $.trim($src.attr('class'));
+    classes = $src.attr('class').trim();
```
Comment (Omerisra6) [inline assets/lib/e-select2/js/e-select2.full.js:5785]:
```suggestion
    classes = $src?.attr('class')?.trim();
```

File: assets/lib/e-select2/js/e-select2.full.js [hunk 5]
```diff
@@ -5767,7 +5769,7 @@ S2.define('select2/compat/utils',[
   function syncCssClasses ($dest, $src, adapter) {
     var classes, replacements = [], adapted;
 
-    classes = $.trim($dest.attr('class'));
+    classes = $dest.attr('class').trim();
```
Comment (Omerisra6) [inline assets/lib/e-select2/js/e-select2.full.js:5772]:
```suggestion
    classes = $dest?.attr('class')?.trim();
```

File: assets/lib/e-select2/js/e-select2.full.js [hunk 6]
```diff
@@ -4931,7 +4933,7 @@ S2.define('select2/defaults',[
 
     function matcher (params, data) {
       // Always return the object if there is nothing to compare
-      if ($.trim(params.term) === '') {
+      if (params.term.trim() === '') {
```
Comment (Omerisra6) [inline assets/lib/e-select2/js/e-select2.full.js:4936]:
```suggestion
      if (params?.term?.trim() === '') {
```

**Review by Omerisra6 — CHANGES_REQUESTED** [review]

File: assets/lib/e-select2/js/e-select2.full.js [hunk 7]
```diff
@@ -4931,7 +4933,7 @@ S2.define('select2/defaults',[
 
     function matcher (params, data) {
       // Always return the object if there is nothing to compare
-      if ($.trim(params.term) === '') {
+      if (params?.term?.trim() === '') {
```
Comment (Omerisra6) [inline assets/lib/e-select2/js/e-select2.full.js:4936]:
```suggestion
      if (! params || ! params.term || params.term.trim() === '') {
```

**Review by rami-elementor — DISMISSED** [review]

File: assets/lib/e-select2/js/e-select2.full.js [hunk 8]
```diff
@@ -5392,7 +5392,7 @@ S2.define('select2/core',[
     });
 
     this.selection.on('focus', function (params) {
-		self.trigger('focus', params);
+		self.focus( params);
```
Comment (mark-elementor) [inline assets/lib/e-select2/js/e-select2.full.js:5395]:
```suggestion
      self.focus(params);
```

File: assets/lib/e-select2/js/e-select2.full.js [hunk 9]
```diff
@@ -4931,7 +4933,7 @@ S2.define('select2/defaults',[
 
     function matcher (params, data) {
       // Always return the object if there is nothing to compare
-      if ($.trim(params.term) === '') {
+      if (! params || ! params.term || params.term.trim() === '') {
```
Comment (mark-elementor) [inline assets/lib/e-select2/js/e-select2.full.js:4936]:
```suggestion
      if ((params?.term ?? '').trim() === '') {
```

File: assets/lib/e-select2/js/e-select2.full.js [hunk 10]
```diff
@@ -5767,7 +5769,7 @@ S2.define('select2/compat/utils',[
   function syncCssClasses ($dest, $src, adapter) {
     var classes, replacements = [], adapted;
 
-    classes = $.trim($dest.attr('class'));
+    classes = $dest?.attr('class')?.trim();
```
Comment (mark-elementor) [inline assets/lib/e-select2/js/e-select2.full.js:5772]:
```suggestion
    classes = ($dest?.attr('class') ?? '').trim();
```

File: assets/lib/e-select2/js/e-select2.full.js [hunk 11]
```diff
@@ -5780,7 +5782,7 @@ S2.define('select2/compat/utils',[
       });
     }
 
-    classes = $.trim($src.attr('class'));
+    classes = $src?.attr('class')?.trim();
```
Comment (mark-elementor) [inline assets/lib/e-select2/js/e-select2.full.js:5785]:
```suggestion
    classes = ($src?.attr('class') ?? '').trim();
```

File: assets/lib/e-select2/js/e-select2.full.js [hunk 12]
```diff
@@ -6098,7 +6100,7 @@ S2.define('select2/compat/matcher',[
     function wrappedMatcher (params, data) {
       var match = $.extend(true, {}, data);
 
-      if (params.term == null || $.trim(params.term) === '') {
+      if (params.term == null || params?.term?.trim() === '') {
```
Comment (mark-elementor) [inline assets/lib/e-select2/js/e-select2.full.js:6103]:
```suggestion
      if (params.term == null || (params?.term ?? '').trim() === '') {
```

**Review by Omerisra6 — APPROVED** [review]

**Pull request merged**